### PR TITLE
Allow multiline input for comments / render comments as markdown

### PR DESCRIPTION
**Pull request by dervoeti**
This removes the 255 character limit from comments, adds support for multiple lines in comments and renders them as markdown in the "show" view an observation log entry. In the observation log list, comments are shortened to 255 characters (plus three dots, if they are longer than 255 characters).

### CODE REVIEW

File: backend/application/core/models.py [hunk 1]
```diff
@@ -544,7 +544,7 @@ class Observation_Log(Model):
     )
     severity = CharField(max_length=12, choices=Severity.SEVERITY_CHOICES, blank=True)
     status = CharField(max_length=16, choices=Status.STATUS_CHOICES, blank=True)
-    comment = CharField(max_length=255)
+    comment = TextField()
```
Comment (StefanFl) [inline backend/application/core/models.py:547]: All other fields have a limitation of the length, I would like to see that here as well. Shouldn't 2048 or 4096 characters be enough for a comment?

File: backend/application/core/api/serializers_observation.py [hunk 2]
```diff
@@ -414,11 +414,11 @@ class ObservationAssessmentSerializer(Serializer):
         required=False,
         allow_blank=True,
     )
-    comment = CharField(max_length=255, required=True)
+    comment = CharField(required=True)
```
Comment (StefanFl) [inline backend/application/core/api/serializers_observation.py:417]: see remark about the size limitiation in the model

File: backend/application/core/api/serializers_observation.py [hunk 3]
```diff
@@ -414,11 +414,11 @@ class ObservationAssessmentSerializer(Serializer):
...
 
 
 class ObservationRemoveAssessmentSerializer(Serializer):
-    comment = CharField(max_length=255, required=True)
+    comment = CharField(required=True)
```
Comment (StefanFl) [inline backend/application/core/api/serializers_observation.py:421]: see remark about the size limitiation in the model

File: backend/application/core/api/serializers_observation.py [hunk 4]
```diff
@@ -430,7 +430,7 @@ class ObservationBulkDeleteSerializer(Serializer):
 class ObservationBulkAssessmentSerializer(Serializer):
     severity = ChoiceField(choices=Severity.SEVERITY_CHOICES, required=False)
     status = ChoiceField(choices=Status.STATUS_CHOICES, required=False)
-    comment = CharField(max_length=255, required=True)
+    comment = CharField(required=True)
```
Comment (StefanFl) [inline backend/application/core/api/serializers_observation.py:433]: see remark about the size limitiation in the model

File: frontend/src/core/observation_logs/ObservationLogEmbeddedList.tsx [hunk 5]
```diff
@@ -48,7 +56,7 @@ const ObservationLogEmbeddedList = ({ observation }: ObservationLogEmbeddedListP
                     <TextField source="user_full_name" label="User" sortable={false} />
                     <TextField source="severity" emptyText="---" sortable={false} />
                     <TextField source="status" emptyText="---" sortable={false} />
-                    <TextField source="comment" sortable={false} />
+                    <TextField source="comment_shortened" sortable={false} />
```
Comment (StefanFl) [inline frontend/src/core/observation_logs/ObservationLogEmbeddedList.tsx:59]:
```suggestion
                    <TextField source="comment_shortened" sortable={false} label="Comment" />
```

File: frontend/src/core/observation_logs/ObservationLogEmbeddedList.tsx [hunk 6]
```diff
@@ -25,6 +25,14 @@ const ObservationLogEmbeddedList = ({ observation }: ObservationLogEmbeddedListP
         listContext.data = [];
     }
 
+    listContext.data.forEach((element: any) => {
```
Comment (StefanFl) [inline frontend/src/core/observation_logs/ObservationLogEmbeddedList.tsx:28]: We should have this in the `ObservationLogApprovalList` as well.

File: frontend/src/core/observations/ObservationAssessment.tsx [hunk 7]
```diff
@@ -114,7 +114,7 @@ const ObservationAssessment = () => {
                                 choices={OBSERVATION_VEX_JUSTIFICATION_CHOICES}
                             />
                         )}
-                        <TextInputWide source="comment" validate={validate_required_255} />
+                        <TextInputWide multiline={true} source="comment" validate={validate_required} />
```
Comment (StefanFl) [inline frontend/src/core/observations/ObservationAssessment.tsx:117]:
```suggestion
                        <TextInputWide multiline={true} source="comment" validate={validate_required} minRows={3} />
```

and see remark about the size limitiation in the model

File: frontend/src/core/observations/ObservationBulkAssessment.tsx [hunk 8]
```diff
@@ -137,7 +137,7 @@ const ObservationBulkAssessment = (props: ObservationBulkAssessmentButtonProps)
                                 choices={OBSERVATION_VEX_JUSTIFICATION_CHOICES}
                             />
                         )}
-                        <TextInputWide source="comment" validate={validate_required_255} />
+                        <TextInputWide source="comment" validate={validate_required} />
```
Comment (StefanFl) [inline frontend/src/core/observations/ObservationBulkAssessment.tsx:140]:
```suggestion
                        <TextInputWide source="comment" validate={validate_required} minRows={3} />
```

and see remark about the size limitiation in the model

File: frontend/src/core/observations/ObservationRemoveAssessment.tsx [hunk 9]
```diff
@@ -84,7 +84,7 @@ const ObservationRemoveAssessment = () => {
                 <DialogTitle>Observation Remove Assessment</DialogTitle>
                 <DialogContent>
                     <SimpleForm onSubmit={observationUpdate} toolbar={<CustomToolbar />}>
-                        <TextInputWide source="comment" validate={validate_required_255} />
+                        <TextInputWide source="comment" validate={validate_required} />
```
Comment (StefanFl) [inline frontend/src/core/observations/ObservationRemoveAssessment.tsx:87]:
```suggestion
                        <TextInputWide source="comment" validate={validate_required} minRows={3} />
```

and see remark about the size limitiation in the model

**Review by StefanFl — CHANGES_REQUESTED** [review]
It's a great enhancement

**Comment by dervoeti** [comment]
Thanks! Addressed all of your comments.

File: backend/application/core/migrations/0038_alter_observation_log_comment.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ("core", "0037_product_authorization_group_member"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="observation_log",
+            name="comment",
+            field=models.TextField(),
```
Comment (StefanFl) [inline backend/application/core/migrations/0038_alter_observation_log_comment.py:16]: The migration needs to be regenerated.

File: docker/frontend/dev.Dockerfile [hunk 11]
```diff
@@ -0,0 +1,6 @@
+FROM node:20.11.1-alpine3.19@sha256:c0a3badbd8a0a760de903e00cedbca94588e609299820557e72cba2a53dbaa2c as build
```
Comment (StefanFl) [inline docker/frontend/dev.Dockerfile:1]: This file shouldn't be here. It shouldn't be needed anyway. `docker compose build` / `docker compose up` starts the frontend in development mode with hot reloading.

File: frontend/.env.example [hunk 12]
```diff
@@ -0,0 +1,7 @@
+API_BASE_URL=http://localhost:8000/api
```
Comment (StefanFl) [inline frontend/.env.example:1]: This file shouldn't be here.

File: frontend/runtime-env.js [hunk 13]
```diff
@@ -0,0 +1 @@
+window.__RUNTIME_CONFIG__ = {"API_BASE_URL":"http://localhost:8000/api","OIDC_ENABLE":"true","OIDC_AUTHORITY":"http://localhost:8080/realms/secobserve","OIDC_CLIENT_ID":"secobserve","OIDC_REDIRECT_URI":"http://localhost:3000","OIDC_POST_LOGOUT_REDIRECT_URI":"http://localhost:3000"};
```
Comment (StefanFl) [inline frontend/runtime-env.js:1]: This file shouldn't be here.

**Review by StefanFl — CHANGES_REQUESTED** [review]
A few more remarks

**Comment by dervoeti** [comment]
Right, accidentally committed some of my local dev files, I'll streamline my setup later today so I won't need those files anymore. Regenerated the migration as well.

File: frontend/src/core/observations/ObservationBulkAssessment.tsx [hunk 14]
```diff
@@ -137,7 +137,7 @@ const ObservationBulkAssessment = (props: ObservationBulkAssessmentButtonProps)
                                 choices={OBSERVATION_VEX_JUSTIFICATION_CHOICES}
                             />
                         )}
-                        <TextInputWide source="comment" validate={validate_required_255} />
+                        <TextInputWide source="comment" validate={validate_required_4096} minRows={3} />
```
Comment (StefanFl) [inline frontend/src/core/observations/ObservationBulkAssessment.tsx:140]:
```suggestion
                        <TextInputWide source="comment" validate={validate_required_4096} multiline={true} minRows={3} />
```

File: frontend/src/core/observations/ObservationRemoveAssessment.tsx [hunk 15]
```diff
@@ -84,7 +84,7 @@ const ObservationRemoveAssessment = () => {
                 <DialogTitle>Observation Remove Assessment</DialogTitle>
                 <DialogContent>
                     <SimpleForm onSubmit={observationUpdate} toolbar={<CustomToolbar />}>
-                        <TextInputWide source="comment" validate={validate_required_255} />
+                        <TextInputWide source="comment" validate={validate_required_4096} minRows={3} />
```
Comment (StefanFl) [inline frontend/src/core/observations/ObservationRemoveAssessment.tsx:87]:
```suggestion
                        <TextInputWide source="comment" validate={validate_required_4096} multiline={true} minRows={3} />
```

**Review by StefanFl — CHANGES_REQUESTED** [review]
Sorry, 2 minor issues slipped through before

**Comment by dervoeti** [comment]
Fixed :slightly_smiling_face:

**Review by StefanFl — APPROVED** [review]

**Pull request merged**